var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -236,104 +231,60 @@ def libs(self):
-        extra_args = ['--enable-unsupported',
-                      '--enable-symbols=yes']
-
-        # Do not specify the prefix of zlib if it is in a system directory
-        # (see https://github.com/spack/spack/pull/21900).
-        zlib_prefix = self.spec['zlib'].prefix
-        extra_args.append('--with-zlib={0}'.format(
-            'yes' if is_system_path(zlib_prefix) else zlib_prefix))
-
-        extra_args += self.enable_or_disable('threadsafe')
-        extra_args += self.enable_or_disable('cxx')
-        extra_args += self.enable_or_disable('hl')
-        extra_args += self.enable_or_disable('fortran')
-        extra_args += self.enable_or_disable('java')
-        extra_args += self.with_or_without('szip')
-
-        api = self.spec.variants['api'].value
-        if api != 'none':
-            extra_args.append('--with-default-api-version=' + api)
-
-        if self.spec.satisfies('@1.10:'):
-            if '+debug' in self.spec:
-                extra_args.append('--enable-build-mode=debug')
-            else:
-                extra_args.append('--enable-build-mode=production')
-        else:
-            if '+debug' in self.spec:
-                extra_args.append('--enable-debug=all')
-            else:
-                extra_args.append('--enable-production')
+        args = [self.define('ALLOW_UNSUPPORTED', True)]
+        args.append(self.define_from_variant('HDF5_ENABLE_Z_LIB_SUPPORT', 'zlib'))
+        args.append(self.define_from_variant('BUILD_SHARED_LIBS', 'shared'))
+        args.append(self.define_from_variant('ONLY_SHARED_LIBS', 'shared'))
 
-            # '--enable-fortran2003' no longer exists as of version 1.10.0
-            if '+fortran' in self.spec:
-                extra_args.append('--enable-fortran2003')
-            else:
-                extra_args.append('--disable-fortran2003')
+        if '+szip' in spec:
+            args.append(self.define_from_variant(
+                        'HDF5_ENABLE_SZIP_SUPPORT', 'szip'))
+            args.append(self.define('USE_LIBAEC', True))
+            args.append(self.define('HDF5_ENABLE_SZIP_ENCODING', True))
+            args.append(self.define('SZIP_INCLUDE_DIR', 
+                                        spec['libaec'].prefix.include))
+            args.append(self.define('SZIP_DIR', spec['libaec'].prefix.lib)) 
+
+        args.append(self.define_from_variant('HDF5_ENABLE_PARALLEL', 'mpi'))
+
+        args.append(self.define_from_variant('HDF5_ENABLE_THREADSAFE', 'threadsafe'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_HL_LIB', 'hl'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_CPP_LIB', 'cxx'))
 
-        if '+shared' in self.spec:
-            extra_args.append('--enable-shared')
+        args.append(self.define_from_variant('HDF5_BUILD_FORTRAN', 'fortran'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_JAVA', 'java'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_TOOLS', 'tools'))
```

Comment:
I think the extra newlines between these make the configure less readable; I'd actually smash them all into the construction of `args`:
```python
        args = [
            self.define('ALLOW_UNSUPPORTED', True),
            self.define('BUILD_TESTING', self.run_tests),
            self.define_from_variant('HDF5_ENABLE_Z_LIB_SUPPORT', 'zlib'),
            self.define_from_variant('HDF5_ENABLE_SZIP_SUPPORT', 'szip')
            self.define_from_variant('BUILD_SHARED_LIBS', 'shared'),
            self.define_from_variant('ONLY_SHARED_LIBS', 'shared'),
            self.define_from_variant('HDF5_ENABLE_PARALLEL', 'mpi'),
            self.define_from_variant('HDF5_ENABLE_THREADSAFE', 'threadsafe'),
            self.define_from_variant('HDF5_BUILD_HL_LIB', 'hl'),
            self.define_from_variant('HDF5_BUILD_CPP_LIB', 'cxx'),
            self.define_from_variant('HDF5_BUILD_FORTRAN', 'fortran'),
            self.define_from_variant('HDF5_BUILD_JAVA', 'java'),
            self.define_from_variant('HDF5_BUILD_TOOLS', 'tools'),
        ]

        if '+szip' in spec:
            args.append(self.define('USE_LIBAEC', True))
            args.append(self.define('HDF5_ENABLE_SZIP_ENCODING', True))
            args.append(self.define('SZIP_INCLUDE_DIR', 
                                        spec['libaec'].prefix.include))
            args.append(self.define('SZIP_DIR', spec['libaec'].prefix.lib)) 

        api = spec.variants['api'].value
        if api != 'default':
            args.append(self.define('DEFAULT_API_VERSION', api))
```

Reply:
Thank you, done!